sink_service/source/main.c

```diff
@@ -60,17 +61,75 @@ static bool get_service_name(char service_name[MAX_SIZE_SERVICE_NAME], unsigned
     return true;
 }
 
+
+/**
+ * \brief   Obtains environment parameters to control sink settings
+ * \param   bitrate
+ *          Pointer where to store bitrate value (if any)
+ * \param   port_name
+ *          Pointer where to store port_name value (if any)
+ * \param   sink_id
+ *          Pointer where to store sink_id value (if any)
+ * \param   max_poll_fail_duration
+ *          Pointer where to store max_poll_fail_duration value (if any)
+ * \return  Number of acquired parameters
+ */
+static bool get_env_parameters(unsigned long * bitrate, char * port_name, unsigned int * sink_id, unsigned int * max_poll_fail_duration)
+{
+    char *pTmp;
+    char ebuf[MAX_SIZE_PARAMETER];
+    unsigned int acquired = 0;
+
+    if (( pTmp = getenv( "WM_GW_SINK_BITRATE" )) != NULL )
+    {
+        memset(ebuf, 0, sizeof(ebuf));
+        strncpy(ebuf, pTmp, MAX_SIZE_PARAMETER - 1);
+        *bitrate = strtol(ebuf, NULL, 0);
+        LOGI("WM_GW_SINK_BITRATE: %lu\n", *bitrate);
+        acquired += 1;
+    }
+    if (( pTmp = getenv( "WM_GW_SINK_ID" )) != NULL )
+    {
+        memset(ebuf, 0, sizeof(ebuf));
+        strncpy(ebuf, pTmp, MAX_SIZE_PARAMETER - 1);
+        *sink_id=strtoul(ebuf, NULL, 0);
+        LOGI("WM_GW_SINK_ID: %lu\n", *sink_id);
+        acquired += 1;
+    }
+    if (( pTmp = getenv( "WM_GW_SINK_UART_PORT" )) != NULL )
+    {
+        memset(port_name, 0, sizeof(&port_name));
+        strncpy(port_name, pTmp, MAX_SIZE_PARAMETER - 1);
+        LOGI("WM_GW_SINK_UART_PORT: %s\n", port_name);
+        acquired += 1;
+    }
+    if (( pTmp = getenv( "WM_GW_SINK_MAX_POLL_FAIL_DURATION" )) != NULL )
+    {
+        memset(ebuf, 0, sizeof(ebuf));
+        strncpy(ebuf, pTmp, MAX_SIZE_PARAMETER - 1);
+        *max_poll_fail_duration = strtoul(ebuf, NULL, 0);
+        LOGI("WM_GW_SINK_MAX_POLL_FAIL_DURATION: %lu\n", *max_poll_fail_duration);
+        acquired += 1;
+    }
+    return acquired;
```

Comment:
What is the purpose of acquired that is not used by caller?

Reply:
I left this so that the caller could now how many items were decode (and act on it if meaningful).

However! I forgot to change the function's return value :sweat_smile:

Reply:
I still don't really see the purpose of acquired value.
This function is static one so local to this module that doesn't use it.
I prefer to avoid "just in case" code. If needed later, let add it later ;-)

Reply:
Ok, no problem from my side in removing it.